src/main/java/build/buildfarm/instance/stub/StubInstance.java

```diff
@@ -327,34 +328,14 @@ public void execute(
     throw new UnsupportedOperationException();
   }
 
-  private void requeue(Operation operation) {
-    try {
-      ExecuteOperationMetadata metadata =
-          operation.getMetadata().unpack(ExecuteOperationMetadata.class);
-
-      ExecuteOperationMetadata executingMetadata = metadata.toBuilder()
-          .setStage(ExecuteOperationMetadata.Stage.QUEUED)
-          .build();
-
-      operation = operation.toBuilder()
-          .setMetadata(Any.pack(executingMetadata))
-          .build();
-      putOperation(operation);
-    } catch(InvalidProtocolBufferException ex) {
-      // operation is dropped on the floor
-    }
-  }
-
   @Override
-  public void match(Platform platform, boolean requeueOnFailure, Predicate<Operation> onMatch) {
-    Operation operation = operationQueueBlockingStub.get().take(TakeOperationRequest.newBuilder()
+  public void match(Platform platform, InterruptingPredicate<Operation> onMatch) throws InterruptedException {
+    Operation operation = operationQueueBlockingStub.get()
+        .take(TakeOperationRequest.newBuilder()
         .setInstanceName(getName())
         .setPlatform(platform)
         .build());
-    boolean successful = onMatch.test(operation);
-    if (!Thread.currentThread().isInterrupted() && !successful && requeueOnFailure) {
-      requeue(operation);
-    }
+    boolean success = onMatch.test(operation);
```

Comment:
so the return value is not used?

Reply:
nope. the stub client just takes from the queue, with the same semantics of the match interface, and leaves any failure recovery to the caller. publicising the requeue method would extend the instance interface unnecessarily, calling someone else's requeue method would add a mutual dependency for other requeuers. I might be convinced that this deserves a `take` method with runnables instead of predicates, but this was convenient.

Reply:
Switched this to using a runnable for now